### PR TITLE
Create Reviews

### PR DESCRIPTION
**Pull request by johnedelmapa**
PR description: Create Reviews
Content: Reviews for host and guest users
[Commands]
docker-compose exec web rails db:migrate:reset

[Pre-conditions]

List of test user for facebook:
EMAILS - 
harry_htekcwa_tan@tfbnw.net
linda_omtztbw_chu@tfbnw.net
christopher_plhmsdn_lee@tfbnw.net
ONE TIME PASSWORD - 
112198V74778490N

To check the reviews module you need to register multiple accounts. You also need rooms and reservations from different user to test. When you already have reservations or trips. Try to create review for guest or for host. When review is created it should also provided to user and room show page.
	
[Expected Output]
Working reviews integrated with modal and raty jquery. Show page should also works.
[Notes]
Use port: 3002 to check.

### CODE REVIEW

**Comment by ivanraybuglosa** [comment]
1. In your room page, add the review count and average reviews for the room(refer to the link for reference - https://prntscr.com/p0muyn)

2. In your user profile page, only display the reviews from hosts and guests for the user. Don't display the reviews of the user itself.(refer to the link for reference - https://prntscr.com/p0mvyt)

File: app/views/rooms/show.html.erb [hunk 1]
```diff
@@ -120,17 +120,49 @@
...
 
-         <div class="row">
-            <div class="col-md-12">
-                <h3> 0 Reviews <span id="average_rating"></span></h3>
-                <div class="container">
-                     <div class="text-center">
-                         <h4>There are no reviews.</h4>
+          <div class="row">
+            <div class="col-md-12">                
+                <% if @reviews.count == 0 %>
```
Comment (ivanraybuglosa) [inline app/views/rooms/show.html.erb]: you can check the reviews count by `@reviews.any?`

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -23,13 +23,118 @@
 
     <div class="mt-4 col-9">
         <h2><%= @user.fullname%> </h2>
-        <h4 class="mt-4">Listings (0)</h4>
-        <h4 class="text-center">There are no listings.</h4>
+        <h4 class="mt-4">Listings (<%= @rooms.count%>)</h4>
+        <% if @rooms.count == 0 %>
```
Comment (ivanraybuglosa) [inline app/views/users/show.html.erb]: Check rooms value by `@rooms.any?`

File: app/views/reservations/your_trips.html.erb [hunk 3]
```diff
@@ -35,8 +35,50 @@
                                 <span><%= trip.room.user.fullname %></span>
```
Comment (ivanraybuglosa) [inline app/views/reservations/your_trips.html.erb]: Add a link in this user fullname to redirect to user profile

File: app/views/reservations/your_trips.html.erb [hunk 4]
```diff
@@ -3,29 +3,29 @@
...
                     <div class = "row">
                         <% current_user.reservations.each do |trip| %>
                             <div class="col-2 mt-3 text-muted">
                                 <h6><%= trip.start_date.strftime("%v") %></h6>
                             </div>
 
-                            <div class="col-2 mt-3">
+                             <div class="col-2 mt-3">
                                 <% unless trip.room.photos.blank? %>
                                     <%= image_tag trip.room.photos.first.image.url(:thumb), class: "img-thumbnail" %>
```
Comment (ivanraybuglosa) [inline app/views/reservations/your_trips.html.erb]: add a link for a room in this image to redirect to room page..

File: app/views/rooms/show.html.erb (hunk 1 above)
Comment (johnedelmapa) [reply app/views/rooms/show.html.erb]: This wont work on my side.

File: app/views/users/show.html.erb (hunk 2 above)
Comment (johnedelmapa) [reply app/views/users/show.html.erb]: This wont work on my side.

File: app/views/rooms/show.html.erb [hunk 5]
```diff
@@ -120,17 +120,68 @@
...
+                    <% @reviews.each do |review| %>
+                        <hr/>
+                        <div class = "row">
+                            <div class="col-4">
+                                <% if review.guest.image %>
+                                    <%= image_tag review.guest.image, class: "rounded-circle", style: 'height:50px;width:auto;' %>
+                                <% else %>
+                                    <%= image_tag gravatar_for(review.guest), class: "rounded-circle", style: 'height:50px;width:auto;' %>
+                                <% end %>
+                                <strong><%= review.guest.fullname %></strong>
```
Comment (ivanraybuglosa) [inline app/views/rooms/show.html.erb]: insert a link_to to your fullname

File: app/views/users/show.html.erb [hunk 6]
```diff
@@ -23,13 +23,128 @@
...
+            <% @guests.each do |g| %>
+                <hr/>
+                <div class = "row">
+                    <div class="col-3">
+                        <% if g.guest.image %>
+                            <%= image_tag g.guest.image, class: "rounded-circle", style: 'height:50px;width:auto;' %>
+                        <% else %>
+                            <%= image_tag gravatar_for(g.guest), class: "rounded-circle", style: 'height:50px;width:auto;' %>
+                        <% end %>
+                        <strong><%= g.guest.fullname %></strong>
```
Comment (ivanraybuglosa) [inline app/views/users/show.html.erb]: insert link_to to your guest full name

File: app/views/reservations/your_reservations.html.erb [hunk 7]
```diff
@@ -18,26 +18,74 @@
...
+                                            <%= form_for current_user.passive_relationships.new do |f| %>
+
+                                                <%= f.hidden_field :room_id, value: reservation.room.id %>
+                                                <%= f.hidden_field :reservation_id, value: reservation.id %>
+                                                <%= f.hidden_field :guest_id, value: reservation.user.id %>
+                                                <%= f.hidden_field :review_type, value: 0 %>
+
+
+                                                 <div class="center form-group">
+                                                   <div id="star-ratings_<%= reservation.id %>"></div>
```
Comment (ivanraybuglosa) [inline app/views/reservations/your_reservations.html.erb]: center the stars

File: app/views/reservations/your_trips.html.erb [hunk 8]
```diff
@@ -3,40 +3,87 @@
...
+                                </div>
+                                <div class="modal-body">
+                                    <%= form_for current_user.active_relationships.new do |f| %>
+                                    <%= f.hidden_field :room_id, value: trip.room.id %>
+                                    <%= f.hidden_field :reservation_id, value: trip.id %>
+                                    <%= f.hidden_field :host_id, value: trip.room.user.id %>
+                                    <%= f.hidden_field :review_type, value: 1 %>
+
+                                         <div class="center form-group">
+                                            <div id="star-ratings_<%= trip.id %>"></div>
```
Comment (ivanraybuglosa) [inline app/views/reservations/your_trips.html.erb]: center the stars

**Pull request merged**